### DISCUSSION TITLE
How can I get all the output files in a single, flat directory?

### DISCUSSION THREAD

**Discussion by fregante** [parcel-bundler/parcel#5858] [Q&A]
Basically the opposite of https://github.com/parcel-bundler/parcel/issues/233

Parcel v1 used to output all the files in a single folder, but Parcel v2 seems to retain the files’ original folders, which in some cases result in unsightly paths like the `node_modules` one here:

```js
distribution/manifest.json                                                        565 B    190ms
distribution/icon.png                                                           1.14 KB     98ms
distribution/__/node_modules/webextension-polyfill/dist/browser-polyfill.js    41.05 KB    318ms
distribution/background.js                                                     33.61 KB    318ms
distribution/options.html                                                       1.05 KB    689ms
distribution/options.ed4d9118.css                                                 906 B    690ms
distribution/options.a7e143c4.js                                               34.95 KB    693ms
```

By the way, that file wasn't bundled because I'm requiring directly from HTML via

```html
<script src="~/node_modules/webextension-polyfill"></script>
``` 

Search keys: discard tree folder folders subfolders sub

Related to https://github.com/parcel-bundler/parcel/issues/5808

**Comment by mischnic** [comment]
> __/node_modules/webextension-polyfill/dist/

The question is whether that's a bug or actually the indended behaviour 😄 . I'd say it's a bug (it's just a consequence of the rule "subpaths to entry files are preserved by taking the relative path to the entry root).

You could totally write a custom namer plugin that has different rules

**Reply by fregante** [reply]
> it's just a consequence of the rule "subpaths to entry files are preserved

Indeed, I was hoping that there was a way to undo that without writing a custom namer. Unlike most people in that issue I liked Parcel flat output. This also applies to other paths like `images/icons/blah.svg` and not just to the mentioned bug.

**Reply by Muz2258** [reply]
@mischnic it will be very helpful to have a flag that allows someone to choose between keeping the folder structure and flattening it rather than creating a custom namer.

**Comment by ebrensi** [comment]
I did it by basically copying `parcel/DefaultNamer` and commenting out https://github.com/parcel-bundler/parcel/blob/e0440dcbe6d6e419a412c8c568d94715cdb4282a/packages/namers/default/src/DefaultNamer.js#L124

I actually edited the "packaged" version of it and created a dummy `package.json` for it.  It seems to work ok.  I don't like such a hack solution but it works.
